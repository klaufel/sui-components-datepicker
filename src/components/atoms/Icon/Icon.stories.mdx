import {Meta, Props, Story, Preview} from '@storybook/addon-docs/blocks'
import Icon from '.'

<Meta title="Atoms/Icon" component={Icon} />

# Icon

Let's define a story for our `Icon` component:

<Preview>
  <Story name="default">
    <Icon name="chevronUp" />
  </Story>
</Preview>

## Props

<Props of={Icon} />

## All icons

<Preview>
  <Story name="chevronUp">
    <><Icon name="chevronUp" /><br /><small>chevronUp</small></>
  </Story>
  <Story name="chevronRight">
    <><Icon name="chevronRight" /><br /><small>chevronRight</small></>
  </Story>
  <Story name="chevronDown">
    <><Icon name="chevronDown" /><br /><small>chevronDown</small></>
  </Story>
  <Story name="chevronLeft">
    <><Icon name="chevronLeft" /><br /><small>chevronLeft</small></>
  </Story>
</Preview>
