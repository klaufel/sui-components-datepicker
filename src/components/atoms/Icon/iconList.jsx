import React from 'react'

export const iconList = {
  chevronUp: <path d="M20.4 15.5L12 7.1l-8.4 8.4L5 16.9l7-7 7 7z" />,
  chevronRight: <path d="M8.5 20.4l8.4-8.4-8.4-8.4L7.1 5l7 7-7 7z" />,
  chevronDown: <path d="M20.4 8.5L12 16.9 3.6 8.5 5 7.1l7 7 7-7z" />,
  chevronLeft: <path d="M15.5 20.4L7.1 12l8.4-8.4L16.9 5l-7 7 7 7z" />
}

export const iconNames = Object.keys(iconList)
