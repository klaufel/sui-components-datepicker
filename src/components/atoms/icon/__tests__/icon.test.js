import React from 'react'
import {render} from '@testing-library/react'
import Icon, {ICON_NAMES, ICON_SIZES} from '..'

const IconNames = () => ICON_NAMES.map(name => <Icon key={name} name={name} />)

const IconSizes = () =>
  Object.keys(ICON_SIZES).map(size => (
    <Icon key={size} name={ICON_NAMES[0]} size={size} />
  ))

describe('Icon', () => {
  test('it works with all icons names', () => {
    const {container} = render(<IconNames />)
    expect(container).toMatchSnapshot()
  })

  test('it works with all icons sizes', () => {
    const {container} = render(<IconSizes />)
    expect(container).toMatchSnapshot()
  })
})
