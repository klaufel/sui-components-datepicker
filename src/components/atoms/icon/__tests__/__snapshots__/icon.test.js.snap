// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon it works with all icons names 1`] = `
<div>
  <svg
    class="Icon Icon--md"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4 15.5L12 7.1l-8.4 8.4L5 16.9l7-7 7 7z"
    />
  </svg>
  <svg
    class="Icon Icon--md"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.5 20.4l8.4-8.4-8.4-8.4L7.1 5l7 7-7 7z"
    />
  </svg>
  <svg
    class="Icon Icon--md"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4 8.5L12 16.9 3.6 8.5 5 7.1l7 7 7-7z"
    />
  </svg>
  <svg
    class="Icon Icon--md"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.5 20.4L7.1 12l8.4-8.4L16.9 5l-7 7 7 7z"
    />
  </svg>
</div>
`;

exports[`Icon it works with all icons sizes 1`] = `
<div>
  <svg
    class="Icon Icon--sm"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4 15.5L12 7.1l-8.4 8.4L5 16.9l7-7 7 7z"
    />
  </svg>
  <svg
    class="Icon Icon--md"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4 15.5L12 7.1l-8.4 8.4L5 16.9l7-7 7 7z"
    />
  </svg>
  <svg
    class="Icon Icon--lg"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4 15.5L12 7.1l-8.4 8.4L5 16.9l7-7 7 7z"
    />
  </svg>
  <svg
    class="Icon Icon--xl"
    enable-background="new 0 0 24 24"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4 15.5L12 7.1l-8.4 8.4L5 16.9l7-7 7 7z"
    />
  </svg>
</div>
`;
